lib/helpbase.gi

```diff
@@ -1013,13 +1013,74 @@ InstallGlobalFunction(HELP_GET_MATCHES, function( books, topic, frombegin )
   return [exact, match];
 end);
 
+if not IsBound( InitialSubstringUTF8String ) then
+  InitialSubstringUTF8String:= "dummy";
+fi;
+if not IsBound( LETTERS ) then
+  LETTERS:= "dummy";
+fi;
+if not IsBound( WidthUTF8String ) then
+  WidthUTF8String:= "dummy";
+fi;
+
+BindGlobal( "InitialSubstringUTF8Text", function( str, cols )
+    local esc, parts, len, res, j, pos, word, w;
+
+    # Collect printable parts up to visible length 'cols',
+    # and all "escape sequences" in 'str'
+    # (sequences starting with ESC and stopping with the first letter
+    # afterwards).
+    esc:= CHAR_INT(27);
+    parts:= [];
+    len:= Length( str );
+    res:= "";
+    j:= 0;
+    while true do
+      pos:= Position( str, esc, j );
+      if pos = fail then
+        pos:= len+1;
+      fi;
+      word:= str{ [ j+1 .. pos-1 ] };
+      w:= WidthUTF8String( word );
+      if w <= cols then
+        Append( res, word );
+        cols:= cols - w;
+      elif cols > 0 then
+        Append( res, InitialSubstringUTF8String( word, cols ) );
+        cols:= 0;
+      fi;
+      if len < pos then
+        break;
+      fi;
+      j:= pos + 1;
+      while j <= len and not str[j] in LETTERS do
+        j:= j+1;
+      od;
+      if j > len then
```

Comment:
This is very subjective, but personally I'd find this easier to quickly grasp if it used `PositionProperty`, to match the first search
```suggestion
      j:= PositionProperty(str, c -> c in LETTERS, pos);
      if j = fail then
```